src/mon/OSDMonitor.cc

```diff
@@ -2089,6 +2089,8 @@ void OSDMonitor::encode_pending(MonitorDBStore::TransactionRef t)
   // health
   health_check_map_t next;
   tmp.check_health(cct, &next);
+  // OSD_FILESTORE
+  check_for_filestore_osds(&next);
```

Comment:
I wonder if we should make this check less frequent, for efficiency reasons, maybe only after an upgrade or something?

Reply:
All the other health checks are in check_health(), so for consistency it would be nice if the code in check_for_filestore_osds() is put at the end of check_health().

Reply:
In order the check this less frequently, the current state including the list of filestore OSDs must saved.  When not rechecking the saved state would be used to generate the health information.  Something should determine when OSD(s) have changed state to trigger a recheck.  And when MON initializes the initial filestore OSD state is checked and saved.

Reply:
Ack @neha-ojha @dzafman. Let me do the changes accordingly.

Reply:
@dzafman To get OSD objectstore type, we need to extract it from mon store. To add code at the end of `OSDMap::check_health` to check for filestore OSDs requires [1] passing monstore handle to `OSDMap::check_health` along with adding supporting functions or [2] Other way would be saving osd objectstore type in osdmap which does not look appropriate here. Should I go ahead with implementing [1] option here ?

Reply:
> All the other health checks are in check_health(), so for consistency it would be nice if the code in check_for_filestore_osds() is put at the end of check_health().

@dzafman To get OSD objectstore type, we need to extract it from mon store. To add code at the end of OSDMap::check_health to check for filestore OSDs requires [1] passing monstore handle to OSDMap::check_health along with adding supporting functions or [2] Other way would be saving osd objectstore type in osdmap which does not look appropriate here. Should I go ahead with implementing [1] option here ?